compat/mingw.c

```diff
@@ -383,6 +383,21 @@ static wchar_t *normalize_ntpath(wchar_t *wbuf)
 	return wbuf;
 }
 
+static int do_unlink(const char *pathname, const wchar_t *wpathname,
+		     int (*unlinker)(const wchar_t *))
+{
+	char temppath[MAX_PATH];
+	wchar_t wtemppath[MAX_LONG_PATH];
+	strncpy(temppath, pathname, sizeof(temppath) - 8);
+	strcat(temppath, ".XXXXXX");
+	mktemp(temppath);
```

Comment:
The main problem with the previous revision was that I called `mkstemp` here, which creates and opens the temporary file. So nothing could be renamed to it later...